onionperf/util.py

```diff
@@ -133,8 +133,10 @@ def get_ip_address():
     try:
         data = urllib.urlopen('https://check.torproject.org/').read()
         ip_address = find_ip_address_url(data)
-    except:
-        pass
+    except IOError:
+        logging.warning("DNS failed to resolve, could not retrieve your public IP address. This will affect measurements unless your machine has a public IP address. Falling back to finding your IP locally...")
+    except Exception as e:
```

Comment:
What other exceptions should there be? Blindly catching all exceptions isn't a great way to notice unexpected problems. Things that you can recover from are warnings, unexpected things are bugs and need to be noticed. Maybe they are warnings, maybe fatal errors.

Reply:
All network and dns misconfiguration (no nameserver or no default route etc) will result in an IOError - everything else should indeed be noticed.